AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -749,16 +755,37 @@ - (BOOL)shouldBatchFetch
   }
 }
 
-- (void)handleBatchFetchScrollingToOffset:(CGPoint)targetOffset
+- (void)_scheduleCheckForBatchFetching
 {
-  ASDisplayNodeAssert(_batchContext != nil, @"Batch context should exist");
-  
-  if (![self shouldBatchFetch]) {
+  // Push this to the next runloop to be sure the UITableView has the right content size
```

Comment:
UICollectionView.  But we should probably update this to be more generic so the comment can remain the same in table as well.

Can this logic be moved into the shared area and the methods defined in the protocol for batch fetching scroll view? Even if there are four or five methods in the protocol, if the logic is indeed identical between table and collection, and if the methods seem to make sense to share in this way which at first look it appears that they may be a good fit for, this would certainly be a Great cleanup of the code and help move additional complexity out of these very large files.